src/network/network.cpp

```diff
@@ -713,40 +713,87 @@ class TCPClientConnecter : TCPConnecter {
 	}
 };
 
-void NetworkClientConnectGame(const std::string &connection_string, CompanyID default_company, const char *join_server_password, const char *join_company_password)
+/**
+ * Join a client to the server at with the given connection string.
+ * The default for the passwords is \c nullptr. When the server or company needs a
+ * password and none is given, the user is asked to enter the password in the UI.
+ * This function will return false whenever some information required to join is not
+ * correct such as the company number or the client's name, or when there is not
+ * networking avalabile at all. If the function returns false the connection with
+ * the existing server is not disconnected.
+ * It will return true when it starts the actual join process, i.e. when it
+ * actually shows the join status window.
+ *
+ * @param connection_string     The IP address, port and company number to join as.
+ * @param default_company       The company number to join as when none is given.
+ * @param join_server_password  The password for the server.
+ * @param join_company_password The password for the company.
+ * @return Whether the join has started.
+ */
+bool NetworkClientConnectGame(const std::string &connection_string, CompanyID default_company, const char *join_server_password, const char *join_company_password)
 {
 	CompanyID join_as = default_company;
 	NetworkAddress address = ParseGameConnectionString(&join_as, connection_string, NETWORK_DEFAULT_PORT);
 
 	if (join_as != COMPANY_NEW_COMPANY && join_as != COMPANY_SPECTATOR) {
 		join_as--;
 		if (join_as >= MAX_COMPANIES) {
-			return;
+			return false;
 		}
 	}
 
-	NetworkClientConnectGame(address, join_as, join_server_password, join_company_password);
+	return NetworkClientConnectGame(address, join_as, join_server_password, join_company_password);
 }
 
-/* Used by clients, to connect to a server */
-void NetworkClientConnectGame(NetworkAddress &address, CompanyID join_as, const char *join_server_password, const char *join_company_password)
+/**
+ * Join a client to the server at the given address.
+ * See the overloaded NetworkClientConnectGame for more details.
+ *
+ * @param address               The network address of the server to join to.
+ * @param join_as               The company number to join as.
+ * @param join_server_password  The password for the server.
+ * @param join_company_password The password for the company.
+ * @return Whether the join has started.
+ */
+bool NetworkClientConnectGame(NetworkAddress &address, CompanyID join_as, const char *join_server_password, const char *join_company_password)
```

Comment:
Similar as elsewhere, any chance we can make this into std::string while we are here?

Reply:
Not yet; would add too much interdependency between different PRs of me.